tx/textual/valuerenderer/message_test.go

```diff
@@ -0,0 +1,39 @@
+package valuerenderer
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+
+	"cosmossdk.io/tx/textual/internal/testpb"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+func TestMessageValueRenderer_Format(t *testing.T) {
+	tr := NewTextual()
+	mr := &messageValueRenderer{
+		tr: &tr,
+	}
+	// TODO: add a repeated field to the proto once the renderer for repeated fields is implemented.
+	msg := &testpb.Foo{
+		FullName: "the scanner",
+		ChildBar: &testpb.Bar{
+			BarId:      "goku",
+			PowerLevel: 9001,
+		},
+	}
+
+	buf := &bytes.Buffer{}
+	assert.NoError(t, mr.Format(context.Background(), protoreflect.ValueOf((msg).ProtoReflect()), buf))
+
+	want := strings.TrimPrefix(`
+Foo object
+> Full name: the scanner
+> Child bar: Bar object
+>> Bar id: goku
+>> Power level: 9'001
+`, "\n")
+	assert.Equal(t, want, buf.String())
```

Comment:
A great start! At the very least, think of a few more test cases and use an array of test cases. Try to get full code coverage.